### PR TITLE
 Bug 1842304 - Remove feature flag for history interactor removal

### PR DESCRIPTION
**Pull request by MatthewTighe**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1842304

### CODE REVIEW

File: fenix/app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt [hunk 1]
```diff
@@ -377,18 +330,13 @@ class HistoryFragment : LibraryPageFragment<History>(), UserInteractionHandler,
             true
         }
         R.id.history_search -> {
-            historyInteractor.onSearch()
+            historyStore.dispatch(HistoryFragmentAction.SearchClicked)
```
Comment (MatthewTighe) [inline fenix/app/src/main/java/org/mozilla/fenix/library/history/HistoryFragment.kt:333]: I apparently need to drink more coffee

**Comment by gabrielluong** [comment]
@MatthewTighe I am curious if there is a specific gotchas/learnings that we can document now that we have completed the first refactoring.

**Comment by MatthewTighe** [comment]
> I could imagine a doc that just lays out the bugzilla issues or meta.

@gabrielluong Where do you picture such a doc living? I would worry that it would be forgotten in Drive but I'm not sure it really makes sense to keep it in source either

**Comment by gabrielluong** [comment]
> > I could imagine a doc that just lays out the bugzilla issues or meta.
> 
> @gabrielluong Where do you picture such a doc living? I would worry that it would be forgotten in Drive but I'm not sure it really makes sense to keep it in source either

I had the same thought when I initially wrote it so I used doc a bit loosely here. Maybe it should first be a summary comment in https://bugzilla.mozilla.org/show_bug.cgi?id=1839982. Alternatively, we can edit the original RFC with additions.

**Comment by MatthewTighe** [comment]
I added a bit of a summary to the bug.

**Review by MozillaNoah — APPROVED** [review]
image 

[!](https://media.tenor.com/7YOZcMlOLiAAAAAd/the-simpsons-hank-scorpio.gif)

**Pull request merged**